cfgov/unprocessed/js/modules/FlyoutMenu.js

```diff
@@ -9,31 +9,53 @@ var standardType = require( '../modules/util/standard-type' );
  * FlyoutMenu
  * @class
  *
- * @classdesc Initializes a new FlyoutMenu molecule.
+ * @classdesc Initializes new FlyoutMenu behavior.
+ * As added JS behavior, this is added through HTML data-js-hook attributes.
  *
- * @param {HTMLNode} element
- *   The DOM element within which to search for the molecule.
+ * Structure is:
+ * flyout-menu
+ *   flyout-menu_trigger
+ *   flyout-menu_content
+ *     flyout-menu_alt-trigger
+ *
+ * The alt-trigger is for a back button, which may obscure the first trigger.
+ * The flyout can be triggered three ways: through a click of the trigger or
+ * through the click of an alt-trigger.
+ *
+ * @param {HTMLNode} element - The DOM element to attach FlyoutMenu behavior.
  * @returns {FlyoutMenu} An instance.
  */
-function FlyoutMenu( element ) {
+function FlyoutMenu( element ) { // eslint-disable-line max-statements, no-inline-comments, max-len
 
   var BASE_CLASS = 'flyout-menu';
-  var BASE_SEL = '[' + standardType.JS_HOOK + '=' + BASE_CLASS;
+  var SEL_PREFIX = '[' + standardType.JS_HOOK + '=' + BASE_CLASS;
+
+  var BASE_SEL = SEL_PREFIX + ']';
+  var ALT_TRIGGER_SEL = SEL_PREFIX + '_alt-trigger]';
+  var CONTENT_SEL = SEL_PREFIX + '_content]';
+  var TRIGGER_SEL = SEL_PREFIX + '_trigger]';
 
   // TODO: Update atomic-helpers to support CSS selectors for validity check.
   var _dom = dataHook.contains( element, BASE_CLASS ) ? element : null;
-  if ( !_dom ) _dom = element.parentNode.querySelector( BASE_SEL + ']' );
+  if ( !_dom ) _dom = element.parentNode.querySelector( BASE_SEL );
   if ( !_dom ) { throw new Error( 'Selector not found on passed node!' ); }
 
-  var _triggerDom = _dom.querySelector( BASE_SEL + '_trigger]' );
-  var _contentDom = _dom.querySelector( BASE_SEL + '_content]' );
-  var _altTriggerDom = _dom.querySelector( BASE_SEL + '_alt-trigger]' );
+  var _triggerDom = _dom.querySelector( TRIGGER_SEL );
+  var _contentDom = _dom.querySelector( CONTENT_SEL );
 
-  var _isExpanded = false;
+  if ( !_triggerDom ) { throw new Error( TRIGGER_SEL + ' is missing!' ); }
+  if ( !_contentDom ) { throw new Error( CONTENT_SEL + ' is missing!' ); }
 
-  var _transitionEndEvent = _getTransitionEndEvent( _contentDom );
+  var _altTriggerDom = _dom.querySelector( ALT_TRIGGER_SEL );
```

Comment:
No, the alternative trigger is optional and is checked on https://github.com/cfpb/cfgov-refresh/blob/adds_transitions/cfgov/unprocessed/js/modules/FlyoutMenu.js#L76